### PR TITLE
BREAKING: Partial fix for #12356: Make 2FA sortable in user list

### PR DESCRIPTION
**Pull request by snipe**
The original request wants sorting on 2FA *and* groups. Groups is a little harder here, since we have a many to many relationship between users and groups, so I'm not sure how we'd figure out *which* group out of many that we'd be sorting on. But this at least lets us properly sort on the 2FA fields. 

This is, however a BREAKING change for API users. I suspect this field isn't being used a lot in most integrations, but the field name in the API response is NOW `two_factor_optin` versus `two_factor_activated`. Functionality and logic is the same, but the name is different. I don't imagine this will affect many people, but it *is* a breaking change.

### CODE REVIEW

File: app/Http/Controllers/Api/UsersController.php [hunk 1]
```diff
@@ -208,11 +215,39 @@ public function index(Request $request)
             default:
                 $allowed_columns =
                     [
-                        'last_name', 'first_name', 'email', 'jobtitle', 'username', 'employee_num',
-                        'assets', 'accessories', 'consumables', 'licenses', 'groups', 'activated', 'created_at',
-                        'two_factor_enrolled', 'two_factor_optin', 'last_login', 'assets_count', 'licenses_count',
-                        'consumables_count', 'accessories_count', 'phone', 'address', 'city', 'state',
-                        'country', 'zip', 'id', 'ldap_import', 'remote', 'start_date', 'end_date',
+                        'last_name',
```
Comment (uberbrady) [inline app/Http/Controllers/Api/UsersController.php:218]: This is great! I understand your 'logical' grouping here, but I still would prefer you just alphabetize these. So much easier to "visually-grep".

File: app/Http/Transformers/UsersTransformer.php [hunk 2]
```diff
@@ -56,8 +56,8 @@ public function transformUser(User $user)
                 'permissions' => $user->decodePermissions(),
                 'activated' => ($user->activated == '1') ? true : false,
                 'ldap_import' => ($user->ldap_import == '1') ? true : false,
-                'two_factor_activated' => ($user->two_factor_active()) ? true : false,
                 'two_factor_enrolled' => ($user->two_factor_active_and_enrolled()) ? true : false,
+                'two_factor_optin' => ($user->two_factor_active()) ? true : false,
```
Comment (uberbrady) [inline app/Http/Transformers/UsersTransformer.php:60]: I get that this is a little clearer, but IMHO it's not worth the back-compatibility break, and I would prefer we not do it.

**Review by uberbrady — COMMENTED** [review]
I won't go so far as to force 'request changes' on this, but I can't see the benefit in changing the name of this field. The potential damage just outweighs so much the increased clarity. But  if you want to merge this, I won't complain. Or you can try and sell me and I can give a formal approval, if you want.

File: app/Http/Controllers/Api/UsersController.php (hunk 1 above)
Comment (snipe) [reply app/Http/Controllers/Api/UsersController.php:218]: I do get that, but this is more of a logical grep. We basically present these fields in order of learned priority. I tend to prefer alphabetical myself, but in the case of our index pages (and also because English isn't the only language in the world), we try to correspond these pretty tightly to the Presenters that result in which columns are shown by default. 

TLDR; I get it, but I'm not changing it. It's dumb, but everything else is dumber.

File: app/Http/Transformers/UsersTransformer.php (hunk 2 above)
Comment (snipe) [reply app/Http/Transformers/UsersTransformer.php:60]: I really do get this argument. I have seen no evidence of anyone using this field in the API.

Comment (snipe) [reply app/Http/Transformers/UsersTransformer.php:60]: (And it was wrongly-worded to start with :( )

**Review by uberbrady — APPROVED** [review]
Solid fix, and your explanation is completely fair.
